docs/commands/zed.md

```diff
@@ -167,6 +166,25 @@ all adhere to the consistency semantics of the Zed lake.
 > a server instance of `zed` on the same file system and data consistency will
 > be maintained.
 
+### Locating the Lake
+
+At times you may want the Zed CLI tools to access the same lake storage
+used by other tools such as [Zui](https://zui.brimdata.io/). To help
+enable this by default while allowing for separate lake storage when desired,
+`zed` checks each of the following in order to attempt to locate an existing
+lake.
+
+1. The contents of the `-lake` option (if specified)
+2. The contents of the `ZED_LAKE` environment variable (if defined)
+3. A Zed lake service running locally at `http://localhost:9867` (if a socket
+   is listening at that port)
```

Comment:
Maybe tighten this?
```suggestion
3. A Zed lake service running at `http://localhost:9867`
```

Reply:
@nwt: I actually kept this one un-tight intentionally. During black box testing I noticed that if I run a simple program that just listens on `9867` and accepts the incoming connection, and the lake location logic makes it to this step, `zed` effectively hangs indefinitely. I guess it's a hair-split to think that the text as I wrote it communicates this subtlety. But I was trying to express that all it takes is a listening socket for it to commit to this step.

```
#!/usr/local/bin/python3
import socket
import sys

if (len(sys.argv) != 2 or not sys.argv[1].isdigit()):
  print('Usage: listen <port>')
  exit()

p = int(sys.argv[1])
l = []
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.bind(('', p))
s.listen(1)
while 1:
  (c, a) = s.accept()
  l.append(c)
  print('%d: connection from %s' % (len(l), a))
```

Reply:
@philrz: That's a bug!

Reply:
@nwt: Cool! Glad we discussed. I've just opened https://github.com/brimdata/zed/issues/4824 to track this.